lib/datadog/core/configuration/settings.rb

```diff
@@ -475,6 +475,15 @@ def initialize(*_)
             o.default { env_to_bool(Core::Telemetry::Ext::ENV_ENABLED, true) }
             o.lazy
           end
+
+          # The interval in seconds when telemetry must be sent.
+          #
+          # @default `DD_TELEMETRY_HEARTBEAT_INTERVAL` environment variable, otherwise `60`.
+          # @return [Float]
```

Comment:
Would it make sense to document here that is "For internal use only." like we do for some of the other settings?

Additionally, one pattern I've been using for profiling is to shove weird settings into an "advanced" group to make it more clear that usually you shouldn't/don't need to mess with those.